patches/api/0420-Brigadier-based-command-API.patch

```diff
@@ -0,0 +1,746 @@
++        return PROVIDER.signedMessage();
++    }
++
++
++    /**
++     * A component argument.
++     *
++     * @return argument
++     */
++    @NotNull
++    public static ArgumentType<Component> component() {
++        return PROVIDER.component();
++    }
++
++    @ApiStatus.Internal
++    interface VanillaArgumentProvider {
++
++        @NotNull
++        ArgumentType<ItemStack> itemStack();
++
++        @NotNull
++        ArgumentType<BlockState> blockState();
++
++        @NotNull
++        ArgumentType<MessageArgumentResponse> signedMessage();
++
++        @NotNull
++        ArgumentType<UUID> uuid();
++
++        @NotNull
++        ArgumentType<ArgumentResolver<Entity>> entity();
++
++        @NotNull
++        ArgumentType<ArgumentResolver<Player>> player();
++
++        @NotNull
++        ArgumentType<ArgumentResolver<Collection<Entity>>> entities();
++
++        @NotNull
++        ArgumentType<ArgumentResolver<Collection<Player>>> players();
++
++        @NotNull
++        ArgumentType<ArgumentResolver<Collection<PlayerProfile>>> playerProfiles();
++
++        @NotNull
++        ArgumentType<ArgumentResolver<BlockPosition>> blockPos();
++
++        @NotNull
++        ArgumentType<Component> component();
++
++        void validatePrimitiveType(@NotNull ArgumentType<?> type);
++    }
++
++}
+diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
+index 1ab3d8aee3101ca08d9adf5d9ff9f83de911594c..24fa91a9b54dca3e97e37a9da677c2b5d0e42550 100644
+--- a/src/main/java/org/bukkit/Bukkit.java
++++ b/src/main/java/org/bukkit/Bukkit.java
+@@ -2374,6 +2374,7 @@ public final class Bukkit {
+      * @return the default message
+      */
+     @NotNull
++    @Deprecated // Used only for bukkit commands
+     public static net.kyori.adventure.text.Component permissionMessage() {
+         return server.permissionMessage();
+     }
+@@ -2494,6 +2495,11 @@ public final class Bukkit {
+     public static @NotNull org.bukkit.potion.PotionBrewer getPotionBrewer() {
+         return server.getPotionBrewer();
+     }
++
++    @NotNull
++    public static com.mojang.brigadier.CommandDispatcher<io.papermc.paper.command.brigadier.CommandSourceStack> getCommandDispatcher() {
++        return server.getCommandDispatcher();
++    }
+     // Paper end
+ 
+     // Paper start - Folia region threading API
+diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
+index 96a267a676b41dc10f7b18ead826e45c5f6db425..abc177313757002f67d5b100788a0da2a5f2da44 100644
+--- a/src/main/java/org/bukkit/Server.java
++++ b/src/main/java/org/bukkit/Server.java
+@@ -2069,6 +2069,7 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
+      *
+      * @return the default message
+      */
++    @Deprecated // Paper - only used for bukkit commands
+     @NotNull net.kyori.adventure.text.Component permissionMessage();
```

Comment:
I think maybe the `Obsolete` annotation is better here, either way, you should add that comment to the javadoc instead of just a source comment.